src/kasn1.rs

```diff
@@ -123,3 +123,44 @@ impl<'a> asn1::SimpleAsn1Writable for DerEncOctetString<'a> {
         dest.push_slice(self.as_bytes())
     }
 }
+
+type Version = u64;
+
+#[derive(asn1::Asn1Read, asn1::Asn1Write)]
+struct Attribute<'a> {
+    attribute_type: asn1::ObjectIdentifier,
+    attribute_value: asn1::SetOf<'a, asn1::Tlv<'a>>, // ANY
+}
+
+type Attributes<'a> = asn1::SetOf<'a, Attribute<'a>>;
+
+#[derive(asn1::Asn1Read, asn1::Asn1Write)]
+pub struct PrivateKeyInfo<'a> {
+    version: Version,
+    private_key_algorithm: asn1::ObjectIdentifier,
+    private_key: DerEncOctetString<'a>,
+    #[explicit(1)]
+    attributes: Option<Attributes<'a>>,
+}
+
+impl PrivateKeyInfo<'_> {
+    pub fn new<'a>(
+        private_key_asn1: &'a Vec<u8>,
```

Comment:
Why not pass directly a slice here?

Reply:
I think I got confused with the lifetimes. Changed to use the `&[u8]`. Its indeed simpler.